netkan/tests/cli.py

```diff
@@ -45,7 +45,7 @@ def tearDown(self):
 
     def test_clean_all(self):
 
-        result = self.runner.invoke(clean_cache, ['--days', '42'])
+        result = self.runner.invoke(clean_cache, ['--days', '42', '--cache', str(self.cache_path)])
 
```

Comment:
Do we definitely need to call string here? `Path` _should_ do the RightThing&trade; in most circumstances.

Reply:
I think this is now irrelevant with the change to `TemporaryDirectory`.
But I don't think it would have converted to a string implicitly, because this test doesn't:
```python
>>> print(['1', '2', Path('3')])
['1', '2', PosixPath('3')]
```
Or do you mean click would call `str()` on everything non-string when running `invoke()` down the line?

Reply:
I agree, `TemporaryDirectory.name` will be a string. As for your example calling  the `__str__` method on a list won't be calling the `__str__` methods on the members of that list, rather the `__repr__` method.
```
In [4]: for p in ['1', '2', Path('3')]: 
   ...:     print(p) 
   ...:                                                                                                                                                                                                                                                                                                                      
1
2
3
```
So it will probably depend on what happens inside of invoke.

Reply:
That's interesting, thanks!